quad/src/constants.rs

```diff
@@ -32,6 +32,21 @@ pub fn add_res(v: &mut [f64], w: &[f64]) {
     }
 }
 
+pub fn points_transformed(mut points: Vec<f64>, a: f64, b: f64) -> Vec<f64> {
+    points.sort_by(|a, b| a.partial_cmp(b).unwrap());
+    let mut points_transformed = vec![0.0; 0];
+    for point in &points {
+        if b == f64::INFINITY && a.is_finite() {
+            points_transformed.push(1.0 / (*point - a + 1.0));
+        } else if a == f64::NEG_INFINITY && b.is_finite() {
+            points_transformed.push(1.0 / (b - *point + 1.0));
+        } else if a == f64::NEG_INFINITY && b == f64::INFINITY {
+            points_transformed.push(point.signum() / (point.abs() + 1.0));
+        }
```

Comment:
I know I start being annoying, but leveraging the fact that Rust has conditional expressions, you can also lift `points_transformed.push()` out of the `if`, and just keep only the function itself inside :)